src/parsers/manifest/smooth/index.ts

```diff
@@ -469,12 +476,12 @@ function createSmoothStreamingParser(
       const initSegmentInfos = {
         bitsPerSample: representation.bitsPerSample,
         channels: representation.channels,
-        codecPrivateData: representation.codecPrivateData,
+        codecPrivateData: representation.codecPrivateData || "",
         packetSize: representation.packetSize,
         samplingRate: representation.samplingRate,
         protection,
       };
-      representation.index = new RepresentationIndex(index, initSegmentInfos);
+      representation.index = new RepresentationIndex(repIndex, initSegmentInfos);
```

Comment:
I thingk this is actually the occasion to define what RepresentationIndex accepts (for now, it is ``any``)